libraries/Microsoft.Bot.Schema/IActivity.cs

```diff
@@ -158,5 +158,29 @@ public interface IActivity
         /// Returns IMessageDeleteActivity if this is a message delete activity, null otherwise
         /// </summary>
         ISuggestionActivity AsSuggestionActivity();
+
```

Comment:
I would suggest actually _not_ adding these methods to the interface. They are essentially helper methods whose implementation would never need to be substituted because they are acting in matter-of-fact ways on data contracts and, thus, could just be public methods off of the `Activity` class directly.

Broader discussion, which I've mentioned to @cleemullins a couple times, I actually don't think any of the activity stuff should have interfaces since they are essentially representative of data contracts and shouldn't have _any_ behavior what-so-ever.